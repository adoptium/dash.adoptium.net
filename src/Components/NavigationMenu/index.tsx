import React from 'react';
import { useLocation } from 'react-router-dom';
import { Menu } from 'antd';
import { Link } from 'react-router-dom';
import { CloudDownloadOutlined, LineChartOutlined } from '@ant-design/icons';

const NavigationMenu = (): JSX.Element => {
    const location = useLocation();

    const getSelectedKeysByUrl = () => {
        if(location.pathname === '/') return ['1'];
        if(location.pathname === '/download') return ['1'];
        else if(location.pathname === '/trends') return ['2'];
    }

    return (
        <Menu
            mode="inline"
            selectedKeys={getSelectedKeysByUrl()}
            style={{ height: '100%', borderRight: 0, marginTop: 20 }}
        >
            <Menu.Item key="1"><Link to="/download"><CloudDownloadOutlined /> Download</Link></Menu.Item>
            <Menu.Item key="2"><Link to="/trends"><LineChartOutlined /> Trends</Link></Menu.Item>
         </Menu>
    )
}

export default NavigationMenu;
