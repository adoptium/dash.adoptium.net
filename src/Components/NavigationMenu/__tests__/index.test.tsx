import React from 'react';
import { act, render } from '@testing-library/react';
import { vi, describe, expect, test } from 'vitest'
import NavigationMenu from '../index';
import { useLocation } from 'react-router-dom';

vi.mock('react-router-dom');

describe('NavigationMenu component', () => {
    test('renders correctly /download path', async() => {
        vi.mocked(useLocation).mockReturnValue({path: '/download'});

        let container;
        await act(async () => {
            ({ container } = render(
              <NavigationMenu
              />
            ));
          });

          expect(container).toMatchSnapshot();
    });
    
    test('renders correctly /trends path', async () => {
        vi.mocked(useLocation).mockReturnValue({path: '/trends'});

        let container;
        await act(async () => {
            ({ container } = render(
            <NavigationMenu
            />
            ));
        });

      expect(container).toMatchSnapshot();
    });
});
