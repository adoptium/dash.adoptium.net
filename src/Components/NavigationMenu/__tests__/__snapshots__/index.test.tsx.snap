// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu component > renders correctly /download path 1`] = `
<div>
  <ul
    class="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
    data-menu-list="true"
    role="menu"
    style="height: 100%; border-right: 0; margin-top: 20px;"
    tabindex="0"
  >
    <li
      aria-describedby="test-id"
      class="ant-menu-item"
      data-menu-id="rc-menu-uuid-test-1"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        aria-label="cloud-download"
        class="anticon anticon-cloud-download ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="cloud-download"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M624 706.3h-74.1V464c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v242.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.7a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9z"
          />
          <path
            d="M811.4 366.7C765.6 245.9 648.9 160 512.2 160S258.8 245.8 213 366.6C127.3 389.1 64 467.2 64 560c0 110.5 89.5 200 199.9 200H304c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8h-40.1c-33.7 0-65.4-13.4-89-37.7-23.5-24.2-36-56.8-34.9-90.6.9-26.4 9.9-51.2 26.2-72.1 16.7-21.3 40.1-36.8 66.1-43.7l37.9-9.9 13.9-36.6c8.6-22.8 20.6-44.1 35.7-63.4a245.6 245.6 0 0152.4-49.9c41.1-28.9 89.5-44.2 140-44.2s98.9 15.3 140 44.2c19.9 14 37.5 30.8 52.4 49.9 15.1 19.3 27.1 40.7 35.7 63.4l13.8 36.5 37.8 10C846.1 454.5 884 503.8 884 560c0 33.1-12.9 64.3-36.3 87.7a123.07 123.07 0 01-87.6 36.3H720c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h40.1C870.5 760 960 670.5 960 560c0-92.7-63.1-170.7-148.6-193.3z"
          />
        </svg>
      </span>
      <span
        class="ant-menu-title-content"
      />
    </li>
    <li
      aria-describedby="test-id"
      class="ant-menu-item"
      data-menu-id="rc-menu-uuid-test-2"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        aria-label="line-chart"
        class="anticon anticon-line-chart ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="line-chart"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM305.8 637.7c3.1 3.1 8.1 3.1 11.3 0l138.3-137.6L583 628.5c3.1 3.1 8.2 3.1 11.3 0l275.4-275.3c3.1-3.1 3.1-8.2 0-11.3l-39.6-39.6a8.03 8.03 0 00-11.3 0l-230 229.9L461.4 404a8.03 8.03 0 00-11.3 0L266.3 586.7a8.03 8.03 0 000 11.3l39.5 39.7z"
          />
        </svg>
      </span>
      <span
        class="ant-menu-title-content"
      />
    </li>
  </ul>
  <div
    aria-hidden="true"
    style="display: none;"
  />
</div>
`;

exports[`NavigationMenu component > renders correctly /trends path 1`] = `
<div>
  <ul
    class="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
    data-menu-list="true"
    role="menu"
    style="height: 100%; border-right: 0; margin-top: 20px;"
    tabindex="0"
  >
    <li
      aria-describedby="test-id"
      class="ant-menu-item"
      data-menu-id="rc-menu-uuid-test-1"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        aria-label="cloud-download"
        class="anticon anticon-cloud-download ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="cloud-download"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M624 706.3h-74.1V464c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v242.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.7a8 8 0 0012.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9z"
          />
          <path
            d="M811.4 366.7C765.6 245.9 648.9 160 512.2 160S258.8 245.8 213 366.6C127.3 389.1 64 467.2 64 560c0 110.5 89.5 200 199.9 200H304c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8h-40.1c-33.7 0-65.4-13.4-89-37.7-23.5-24.2-36-56.8-34.9-90.6.9-26.4 9.9-51.2 26.2-72.1 16.7-21.3 40.1-36.8 66.1-43.7l37.9-9.9 13.9-36.6c8.6-22.8 20.6-44.1 35.7-63.4a245.6 245.6 0 0152.4-49.9c41.1-28.9 89.5-44.2 140-44.2s98.9 15.3 140 44.2c19.9 14 37.5 30.8 52.4 49.9 15.1 19.3 27.1 40.7 35.7 63.4l13.8 36.5 37.8 10C846.1 454.5 884 503.8 884 560c0 33.1-12.9 64.3-36.3 87.7a123.07 123.07 0 01-87.6 36.3H720c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h40.1C870.5 760 960 670.5 960 560c0-92.7-63.1-170.7-148.6-193.3z"
          />
        </svg>
      </span>
      <span
        class="ant-menu-title-content"
      />
    </li>
    <li
      aria-describedby="test-id"
      class="ant-menu-item"
      data-menu-id="rc-menu-uuid-test-2"
      role="menuitem"
      style="padding-left: 24px;"
      tabindex="-1"
    >
      <span
        aria-label="line-chart"
        class="anticon anticon-line-chart ant-menu-item-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="line-chart"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM305.8 637.7c3.1 3.1 8.1 3.1 11.3 0l138.3-137.6L583 628.5c3.1 3.1 8.2 3.1 11.3 0l275.4-275.3c3.1-3.1 3.1-8.2 0-11.3l-39.6-39.6a8.03 8.03 0 00-11.3 0l-230 229.9L461.4 404a8.03 8.03 0 00-11.3 0L266.3 586.7a8.03 8.03 0 000 11.3l39.5 39.7z"
          />
        </svg>
      </span>
      <span
        class="ant-menu-title-content"
      />
    </li>
  </ul>
  <div
    aria-hidden="true"
    style="display: none;"
  />
</div>
`;
